### PR TITLE
Add Types to useOverlay Context

### PR DESCRIPTION
**Pull request by amrro**
## Description
`useOverlay` controls which Modal should be showing across the app. Types for `OverlayContext` & which modals the apps contains `GovrnModals` has been added.

### CODE REVIEW

File: apps/protocol-frontend/src/components/AttestationsTableShell.tsx [hunk 1]
```diff
@@ -16,10 +16,8 @@ import PageHeading from './PageHeading';
 import AttestationsTable from './AttestationsTable';
 import EmptyContributions from './EmptyContributions';
 import MyAttestationsTable from './MyAttestationsTable';
-import { ModalWrapper } from '@govrn/protocol-ui';
+import ModalWrapper from './ModalWrapper';
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/AttestationsTableShell.tsx:19]: Why move the modal wrapper out of the component libary?

File: apps/protocol-frontend/src/contexts/OverlayContext.tsx [hunk 2]
```diff
@@ -1,42 +1,55 @@
...
+export type OverlayContextType = {
+  modals: GovrnModals;
+  setModals: (modals: Partial<GovrnModals>) => void;
+};
+
+export const OverlayContext = createContext({} as OverlayContextType);
 
 interface OverlayProviderProps {
-  children: any;
+  children?: ReactNode | undefined;
```
Comment (alexkeating) [inline apps/protocol-frontend/src/contexts/OverlayContext.tsx:30]: Nice!

**Review by jonathanprozzi — APPROVED** [review]
Nice -- love the types and changes to the modals structure. I think we can maybe add a presentational wrapper back into the UI library, but having this near the relevant context (in the frontend app package) makes sense.

**Review by jonathanprozzi — APPROVED** [review]
Nice -- love the types and changes to the modals structure. I think we can maybe add a presentational wrapper back into the UI library, but having this near the relevant context (in the frontend app package) makes sense.

**Pull request merged**